src/Mapbender/CoreBundle/Component/LogoutSuccessHandler.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Mapbender\CoreBundle\Component;
+
+use Symfony\Component\Security\Http\Logout\LogoutSuccessHandlerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\Routing\Router;
+
+class LogoutSuccessHandler implements LogoutSuccessHandlerInterface
+{
+
+    protected $router;
+
+    public function __construct(Router $router)
+    {
+            $this->router = $router;
+    }    
+    
+    public function onLogoutSuccess(Request $request)
+    {
+       $referer_url = $request->headers->get('referer');
+       $response = new RedirectResponse($referer_url);		
```

Comment:
Please don't use Request and Response objects inside of Component folder. 
Redirects and Repsonse should  be formed and used inside of bundle Controller folder.